code_folder/D1_model_components.py

```diff
@@ -117,6 +133,66 @@ def transformer_constant_efficiency_fix(model, dict_asset, **kwargs):
         kwargs["transformers"].update({dict_asset["label"]: transformer})
         return
 
+    def transformer_constant_efficiency_multiple_buses_fix(model, dict_asset, **kwargs):
+        """
+
+        Parameters
+        ----------
+        dict_asset:
+        dictionary of the asset
+        kwargs:
+        other parameters, basically the busses components
+
+        Returns
+        -------
+
+        """
+        # check if the transformer has multiple input or multiple outpus busses
+        if isinstance(dict_asset["input_bus_name"], list):
+            inputs = {}
+            index = 0
+            for bus in dict_asset["input_bus_name"]:
+                variable_costs = dict_asset["opex_var"]["value"][index]
+                inputs[kwargs["busses"][bus]] = solph.Flow(
+                    nominal_value=dict_asset["installedCap"]["value"],
+                    variable_costs=variable_costs,
+                )
+                index += 1
+            outputs = {kwargs["busses"][dict_asset["output_bus_name"]]: solph.Flow()}
+            efficiencies = {}
+            for i in range(len(dict_asset["efficiency"]["value"])):
+                efficiencies[kwargs["busses"][dict_asset["input_bus_name"][i]]] = (
+                    1 / dict_asset["efficiency"]["value"][i]
+                )
+
+        else:
+            inputs = {kwargs["busses"][dict_asset["input_bus_name"]]: solph.Flow()}
+            outputs = {}
+            index = 0
+            for bus in dict_asset["output_bus_name"]:
+                variable_costs = dict_asset["opex_var"]["value"][index]
+                outputs[kwargs["busses"][bus]] = solph.Flow(
+                    nominal_value=dict_asset["installedCap"]["value"],
+                    variable_costs=variable_costs,
+                )
+                index += 1
+            efficiencies = {}
+            for i in range(len(dict_asset["efficiency"]["value"])):
+                efficiencies[
+                    kwargs["busses"][dict_asset["output_bus_name"][i]]
+                ] = dict_asset["efficiency"]["value"][i]
+
+        transformer = solph.Transformer(
+            label=dict_asset["label"],
+            inputs=inputs,
+            outputs=outputs,
+            conversion_factors=efficiencies,
+        )
+
+        model.add(transformer)
+        kwargs["transformers"].update({dict_asset["label"]: transformer})
```

Comment:
Wouldnt transformer_constant_efficiency_multiple_buses_fix also work if there are not multiple busses? then it could replace transformer_constant_efficiency_fix.